frontend/libs/studio-components/src/components/StudioGridSelector/StudioGridSelector.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { ChangeEvent, useEffect, useState } from 'react';
+import classes from './StudioGridSelector.module.css';
+import cn from 'classnames';
+
+type StudioGridSelectorProps = {
+  disabled?: boolean;
+  sliderValue?: number;
+  handleSliderChange: (newValue: number) => void;
+};
+
+/**
+ * @component
+ *    Component to select a value within the range 1 to 12. Adapted to select value for grid-property on layout components
+ */
+export const StudioGridSelector = ({
+  disabled = false,
+  sliderValue = 12,
+  handleSliderChange,
+}: StudioGridSelectorProps) => {
+  const [value, setValue] = useState<number>(sliderValue);
+  const gridValues = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+  useEffect(() => {
+    setValue(sliderValue ?? 12);
+  }, [sliderValue]);
+
+  return (
+    <div
+      className={disabled ? cn(classes.sliderContainer, classes.disabled) : classes.sliderContainer}
+    >
+      <input
+        className={disabled ? classes.disabled : undefined}
```

Comment:
Regarding the comment on adding InputProps, we can spread the props as follow:

```suggestion
        <input
        {...inputProps}
        className={cn(inputProps.className, { [classes.disabled]: disabled })}
```

The aforementioned suggestion involves spreading the inputProps and incorporating inputProps.className. This approach allows for the seamless passing of custom className/CSS to the slider, eliminating the need to extend or modify this component later.